src/Catalyst.Core.Modules.Cryptography.BulletProofs/Error.cs

```diff
@@ -0,0 +1,103 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Reflection;
+using System.Resources;
+using Catalyst.Core.Modules.Cryptography.BulletProofs.Exceptions;
+
+namespace Catalyst.Core.Modules.Cryptography.BulletProofs
+{
+    internal static class Error
+    {
+        private static ResourceManager _sResourceManager;
+
+        private static ResourceManager ResourceManager =>
+            _sResourceManager ?? (_sResourceManager =
+                new ResourceManager(typeof(Error).FullName,
+                    typeof(Error).GetTypeInfo().Assembly));
+
+        internal static void ThrowErrorFromErrorCode(string errorCode)
+        {
+            switch (errorCode)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Replace this 'switch' statement with 'if' statements to increase readability.](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4318201)